cmd/crank/beta/render/render.go

```diff
@@ -257,3 +308,35 @@ func SetComposedResourceMetadata(cd resource.Object, xr resource.Composite, name
 	or := meta.AsController(meta.TypedReferenceTo(xr, xr.GetObjectKind().GroupVersionKind()))
 	return errors.Wrapf(meta.AddControllerReference(cd, or), "cannot set composite resource %q as controller ref of composed resource", xr.GetName())
 }
+
+func filterExtraResources(ers []unstructured.Unstructured, selector *fnv1beta1.ResourceSelector) (*fnv1beta1.Resources, error) { //nolint:gocyclo // This is a bit long, but not too complex.
+	out := &fnv1beta1.Resources{}
+	for _, er := range ers {
+		er := er
+		if selector.GetApiVersion() != "" && selector.GetApiVersion() != er.GetAPIVersion() {
```

Comment:
```suggestion
		if s := selector.GetApiVersion(); s != "" && s != er.GetAPIVersion() {
```

Nit: This could work too. Just a thought though, it's fine as is.

That said, why the check for `!= ""`? Do we expect to support empty API versions or kinds in the selector?

Reply:
no, we don't support empty API versions or kinds, it was indeed unnecessary.